src/main/lpfConverter.ts

```diff
@@ -0,0 +1,304 @@
+                {});
+            if (ObjectKeys(titleObj).length) {
+                publication.Metadata.Title = titleObj;
+            }
+
+        }
+    }
+    {
+        const language = w3cManifest.inLanguage;
+        if (typeof language === "string") {
+            publication.Metadata.Language = [language];
+        }
+    }
+    {
+        const raw = `${w3cManifest.datePublished || ""}`;
+        const date = moment(raw);
+        if (date.isValid()) {
+            publication.Metadata.PublicationDate = date.toDate();
+        }
+    }
+    {
+        const raw = `${w3cManifest.dateModified || ""}`;
+        const date = moment(raw);
+        if (date.isValid()) {
+            publication.Metadata.Modified = date.toDate();
+        }
+    }
+    {
+        const iso8601 = `${w3cManifest.duration || ""}`;
+        const second = iso8601DurationsToSeconds(iso8601);
+        if (second > 0) {
+            publication.Metadata.Duration = second;
+        }
+    }
+    {
+        const ressources = w3cManifest.resources as JsonMap;
+        const links = convertW3CpublicationLinksToReadiumManifestLink(ressources);
+        if (links.length) {
+            publication.Links = links;
+        }
+    }
+    {
+        const readingOrder = w3cManifest.readingOrder as JsonMap;
+        const links = convertW3CpublicationLinksToReadiumManifestLink(readingOrder);
+        if (links.length) {
+            publication.Spine = links;
+            // https://github.com/readium/r2-shared-js/blob/develop/src/models/publication.ts#L63
+        }
+    }
+
+    const publicationJson = TaJsonSerialize<R2Publication>(publication);
+
+    return publicationJson;
+}
+
+function convertW3CpublicationLinksToReadiumManifestLink(ressources: JsonMap) {
+
+    const linkArray = new Array<Link>();
+
+    if (Array.isArray(ressources)) {
+
+        ressources.forEach(
+            (ressource: any) => {
+                const link = new Link();
+
+                {
+                    const url = ressource.url;
+                    if (typeof url === "string") {
+                        link.Href = url;
+                    }
+                }
+                {
+                    const encodingFormat = ressource.encodingFormat;
+                    if (typeof encodingFormat === "string") {
+                        link.TypeLink = encodingFormat;
+
+                    } else {
+                        const ext = extname(link.Href);
+                        switch (ext) {
+                            case ".mp3":
+                                link.TypeLink = "audio/mpeg";
+                                break;
+                            case ".wav":
+                                link.TypeLink = "audio/wav";
+                                break;
+                            case ".opus":
+                                link.TypeLink = "audio/ogg";
+                                break;
+                            default:
```

Comment:
I would add `.m4a` => `audio/m4a` as well.

Reply:
Also, just to be on the safe side (I don't expect to encounter audiobooks using these codecs, but...): `.ogg`, `.oga` => `audio/ogg`

Reply:
done :)